packages/web/src/pages/rewards-page/components/ClaimAllPanel.tsx

```diff
@@ -0,0 +1,182 @@
+import { useCallback } from 'react'
+
+import {
+  formatCooldownChallenges,
+  useChallengeCooldownSchedule
+} from '@audius/common/hooks'
+import {
+  Box,
+  Button,
+  Flex,
+  IconArrowRight as IconArrow,
+  IconTokenGold,
+  Paper,
+  PlainButton,
+  Text,
+  useTheme
+} from '@audius/harmony'
+
+import { useModalState } from 'common/hooks/useModalState'
+import { useIsMobile } from 'hooks/useIsMobile'
+import { useWithMobileStyle } from 'hooks/useWithMobileStyle'
+
+import styles from '../RewardsTile.module.css'
+import { messages } from '../messages'
+
+export const ClaimAllPanel = () => {
+  const isMobile = useIsMobile() || window.innerWidth < 1080
+  const wm = useWithMobileStyle(styles.mobile)
+  const { cooldownChallenges, cooldownAmount, claimableAmount, isEmpty } =
+    useChallengeCooldownSchedule({ multiple: true })
+  const claimable = claimableAmount > 0
+  const [, setClaimAllRewardsVisibility] = useModalState('ClaimAllRewards')
+  const { iconSizes } = useTheme()
+
+  const onClickClaimAllRewards = useCallback(() => {
+    setClaimAllRewardsVisibility(true)
+  }, [setClaimAllRewardsVisibility])
+
+  const onClickMoreInfo = useCallback(() => {
+    setClaimAllRewardsVisibility(true)
+  }, [setClaimAllRewardsVisibility])
+
+  const handleClick = useCallback(() => {
+    if (claimable) {
+      onClickClaimAllRewards()
+    } else if (cooldownAmount > 0) {
+      onClickMoreInfo()
+    }
+  }, [claimable, cooldownAmount, onClickClaimAllRewards, onClickMoreInfo])
+
+  if (isMobile) {
+    return (
+      <Paper
+        border='strong'
+        alignItems='center'
+        alignSelf='stretch'
+        justifyContent='space-between'
+        css={{ cursor: 'pointer' }}
+        onClick={handleClick}
+      >
+        <Flex direction='column' alignItems='start' w='100%'>
+          <Flex gap='s' alignItems='center'>
+            {claimable ? (
+              <IconTokenGold
```

Comment:
nit: i'm changing these `IconTokenGold|Silver|etc` components to accept 'l' as string height prop. low-pri but could do the same.